application.py

```diff
@@ -47,50 +59,57 @@ def __init__(self):
         """)
         self.qr_code_label.hide()
         self.layout.addWidget(self.qr_code_label, 0, Qt.AlignBottom)
+
         self.current_video_index = 0
         self.video_list = []
         self.video_data = []
+        self.menu_window = None
+
         self.logs = {
                 "app.load_video_data": "",
                 "app.load_videos": "",
                 "app.play_next_video": "",
                 "app.show_qr_code": "",
             }
+        
         self.video_player.finished.connect(self.fade_out_video) # type: ignore
 
+
     def download_video(self, url, local_video_path: str):
         response = requests.get(url, stream=True)
         if response.status_code == 200:
             logger.info(f"Downloading video {local_video_path}...")
             try:
-                with open(f"{local_video_path}", "wb") as file:
+                with open(f"{local_video_path}.dat.incomplete", "wb") as file:
                     for chunk in response.iter_content(chunk_size=1024*1024):
                         file.write(chunk)
+                os.rename(f"{local_video_path}.dat.incomplete", local_video_path)
                 logger.info(f"Video {local_video_path} downloaded successfully")
             except Exception as e:
                 logger.critical(f"Video could not be downloaded. Error: {e}")
         else:
             logger.critical(f"Video could not be downloaded. Status code: {response.status_code}")
 
-    def load_video_data(self, video_data):
+    def set_video_data(self, video_data):
+        self.video_data = video_data
         try:
-            self.video_data = video_data
-            self.logs["app.load_video_data"] = "Список видео успешно загружен."
+            self.service.save_json(self.video_data, f'{ASSETS_FOLDER}/data/video_data.json')
```

Comment:
👍Before, I saved it into a file because I had to access it in the `diagnostic_menu`. Now I changed so that video_data is passed through the class directly.